src/github.com/getlantern/lantern-ui/app/js/app.js

```diff
@@ -122,15 +122,20 @@ var app = angular.module('app', [
           return;
         }
 
-        // try to reconnect indefinitely
-        // when the websocket closes
-        $interval(function() {
-          console.log("Trying to reconnect to disconnected websocket");
-          ds = $websocket('ws://' + document.location.host + '/data');
-          ds.onOpen(function(msg) {
-            $window.location.reload();
-          });
-        }, WS_RECONNECT_INTERVAL);
+        // Temporary workaround for https://bugzilla.mozilla.org/show_bug.cgi?id=1192773
+        if (navigator.userAgent.toLowerCase().indexOf('firefox') > -1) {
+          $rootScope.backendIsGone = true;
+          $rootScope.$digest()
+        } else {
+          // Try to reconnect indefinitely when the websocket closes
+          $interval(function() {
```

Comment:
Yes, I see what you mean, and is one of the things I thought could be the cause of this bug. Unfortunately other tests lead me to the conclusion that what triggers the bug is _previously opened connections_, not many of them trying to open at the very same time (I tried that by heavily randomizing this timeout among other ways).

However, this interval is not going to be created each time, because it effectively overrides the onOpen callback, which will reload the web page and reinitialize the whole thing. A sure way of starting from scratch each time :)